portfolio/src/main/java/com/google/sps/servlets/SentimentAnalysisServlet.java

```diff
@@ -0,0 +1,49 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.cloud.language.v1.Document;
+import com.google.cloud.language.v1.LanguageServiceClient;
+import com.google.cloud.language.v1.Document.Type;
+import com.google.cloud.language.v1.Sentiment;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+
+@WebServlet("/sentiment")
+public class SentimentAnalysisServlet extends HttpServlet {
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String message = request.getParameter("message");
+
+Document doc =
+    Document.newBuilder().setContent(message).setType(Document.Type.PLAIN_TEXT).build();
+LanguageServiceClient languageService = LanguageServiceClient.create();
+Sentiment sentiment = languageService.analyzeSentiment(doc).getDocumentSentiment();
+float score = sentiment.getScore();
+languageService.close();
+
+// Output the sentiment score as HTML.
+response.setContentType("text/html;");
+response.getWriter().println("<h1>Sentiment Analysis</h1>");
+response.getWriter().println("<p>You entered: " + message + "</p>");
+response.getWriter().println("<p>Sentiment analysis score: " + score + "</p>");
+response.getWriter().println("<p><a href=\"/\">Back</a></p>");
```

Comment:
This should work, but if this gets any more complicated it would probably be better to have a separate HTML file containing the response which this endpoint can send rather than constructing the markup line-by-line in Java. It might also be a better UX to fetch the sentiment data in the background (using the `fetch` API for example) and update the DOM dynamically with the response rather than redirecting the user to a whole new results page. Feel free to add this as a TODO here.

Reply:
Thank you for the suggestions! I will leave this as a TODO